src/file_reader.cpp

```diff
@@ -201,7 +201,10 @@ const Buffer MultiPartFileReader::get_buffer(offset_t offset, zsize_t size) cons
   } catch(MMapException& e)
 #endif
   {
-    // The range is several part, or we are on Windows.
+    // We cannot do the mmap, for several possible reasons:
+    // - Mmap offset is too big (>4GB on 32 bits)
+    // - The range is several part
+    // - We are on Windows.
```

Comment:
Please look into `mmapReadOnly()` which is `libzim`'s wrapper around `mmap()` - if the `mmap()` syscall fails a `std::runtime_error` (rather than a `MMapException`) is thrown. I think it must be fixed. Then you will have to add another case to this list.

Reply:
I have think about that, but if the mmap fails, the read will probably fail too. (Or if we mmap a too big size, we may allocation a too big size too).

But we can indeed try our chance and crash later if we cannot read.